### DISCUSSION TITLE
Accessing fasttext vectors in spaCy

### DISCUSSION THREAD

**Discussion by sonynavdeep81** [explosion/spaCy#10034] [Help: Coding & Implementations] [answered]
I have converted the **fasttext** vectors into **spacy** format using `init` command. It has been done successfully as shown below;

<img width="473" alt="Screenshot_5" src="https://user-images.githubusercontent.com/38432505/149085999-7863b03f-d2e0-431b-a8c0-a4290a108774.png">

Here it says that I have to set the path of vectors to **E:\Spacy\output_dir** in **config.cfg** file. Now there are two **config.cfg** files, one default file and the other one created in the **E:\Spacy\output_dir** . Which one to update? I have updated the default one and executed the following code;

<img width="156" alt="Screenshot_6" src="https://user-images.githubusercontent.com/38432505/149086491-6d07f631-b5fd-4e4c-a02f-162eb96de5f3.png">

In this code, I am trying to access vectors in it but it is giving the following error;

<img width="727" alt="Screenshot_7" src="https://user-images.githubusercontent.com/38432505/149086604-d350c57c-d4a1-48b6-acc7-db6cdebe5dab.png">

Where am I doing the mistake?

When I am using the following code I am able to see the vectors;

<img width="234" alt="Screenshot_8" src="https://user-images.githubusercontent.com/38432505/149086724-3e222dad-5cd5-43c3-b0cb-28f9d9b0e5ba.png">

Can't I access the vectors without loading the model? What for is vectors inside the **config.cfg** then? Please explain.

**Comment by polm** [comment, marked as answer]
Don't paste screenshots of code, copy and paste text as text.

`spacy.blank` loads a blank model, it doesn't check any config anywhere. You need to load a model to get vectors. 

When you create a config, you can specify the directory vectors in spaCy format are saved in to load, and that will be made available in your model.

**Reply by sonynavdeep81** [reply]
Thanks for the reply and the advise! I will surely take care of that in the future.  
I am not able to understand the last sentence, can you please explain in a bit more detail. Do you mean that if rather than loading a blank model, we load another model such as `'en_core_web_md`', and provide the path to vectors (which are in spacy format) to the parameter `vectors` under `[initialize]` in `config.cfg`, then the static vectors will be loaded in the model?

**Reply by adrianeboyd** [reply]
`en_core_web_md` is already a full pipeline with its own included vectors. If you replace the vectors, you would break all the statistical components like the tagger.

If you create a config with the path to a pipeline in `[initialize.vectors]` (python package name like `en_core_web_md` or path) and then run either `spacy train` or `spacy assemble`, the output will be a full pipeline (you can load it with `spacy.load`) that contains a complete copy of the vectors from the specified vectors pipeline.

**Reply by sonynavdeep81** [reply]
Thanks!
